flocx_market/db/sqlalchemy/api.py

```diff
@@ -0,0 +1,36 @@
+from oslo_utils import uuidutils
+
+from flocx_market.db.orm import orm
+from flocx_market.db.sqlalchemy import models
+
+
+def offer_get(marketplace_offer_id):
+    return models.Offer.query.filter_by(
+        marketplace_offer_id=marketplace_offer_id).first()
```

Comment:
You could also write this, which is a little shorter:

```
return models.Offer.query.get(marketplace_offer_id)
```

As far as I can tell they end up building the same SQL expression.

Reply:
I actually wondered about this too, because OpenStack always uses the "filter_by/first" expression rather then "get". The reason is because they distinguish between an object's ID (primary key) and UUID (human-usable ID). "get" only works on the primary key.

Incidentally, the database models should probably be updated to follow this pattern as well.